frontend/ui/widget/keyboardlayoutdialog.lua

```diff
@@ -0,0 +1,179 @@
+    border_size = Size.border.window,
+
+}
+
+
+function KeyboardLayoutDialog:init()
+    -- Title & description
+    self.title_widget = FrameContainer:new{
+        padding = self.title_padding,
+        margin = self.title_margin,
+        bordersize = 0,
+        TextWidget:new{
+            text = self.title,
+            face = self.title_face,
+            max_width = self.width,
+        }
+    }
+    self.title_bar = LineWidget:new{
+        dimen = Geom:new{
+            w = self.width,
+            h = Size.line.thick,
+        }
+    }
+
+    local buttons = {}
+    local radio_buttons = {}
+    for k, _ in orderedPairs(self.parent.keyboard.lang_to_keyboard_layout) do
+        table.insert(radio_buttons, {
+            {
+            text = Language:getLanguageName(k),
+            checked = self.parent.keyboard:getKeyboardLayout() == k,
+            provider = k,
+            },
+        })
+    end
+
+    table.insert(buttons, {
+        {
+            text = _("Cancel"),
+            callback = function()
+                UIManager:close(self.parent.keyboard_layout_dialog)
+            end,
+        },
+        {
+            text = _("Chose Language"),
+            is_enter_default = true,
+            callback = function()
+                local provider = self.parent.keyboard_layout_dialog.radio_button_table.checked_button.provider
+                self.parent.keyboard:setKeyboardLayout(provider)
+                UIManager:close(self.parent.keyboard_layout_dialog)
+            end,
+        },
+    })
+
+    self.radio_button_table = RadioButtonTable:new{
+        radio_buttons = radio_buttons,
+        width = self.width * 0.9,
+        focused = true,
+        scroll = false,
+        parent = self,
+        face = self.face,
+    }
+
+    -- Buttons Table
+    self.button_table = ButtonTable:new{
+        width = self.width - 2*self.button_padding,
+        button_font_face = "cfont",
+        button_font_size = 20,
+        buttons = buttons,
+        zero_sep = true,
+        show_parent = self,
+    }
+
+    self.dialog_frame = FrameContainer:new{
+        radius = Size.radius.window,
+        bordersize = Size.border.window,
+        padding = 0,
+        margin = 0,
+        background = Blitbuffer.COLOR_WHITE,
+        VerticalGroup:new{
+            align = "center",
+            self.title_widget,
+            self.title_bar,
+            VerticalSpan:new{
+                width = Size.span.vertical_large*2,
+            },
+            CenterContainer:new{
+                dimen = Geom:new{
+                    w = self.title_bar:getSize().w,
+                    h = self.radio_button_table:getSize().h,
+                },
+                self.radio_button_table,
+            },
+            CenterContainer:new{
+                dimen = Geom:new{
+                    w = self.title_bar:getSize().w,
+                    h = Size.span.vertical_large*2,
+                },
+                LineWidget:new{
```

Comment:
This line should be removed unless you intend to add a "make default" type checkbox below, similar to the open with dialog. (Which might not be a bad idea.)